index.js

```diff
@@ -70,283 +70,314 @@ Fs.readdirSync(Path.join(__dirname, 'helpers')).forEach(function (file) {
+    * context to the paper instance for use in the helpers.
+    *
+    * @param {Object} data - Theme settings (configuration)
+    * @param {Object} assembler - Assembler with getTemplates and getTranslations methods.
+    *
+    */
+    static createInstance(data, assembler) {
+        let translator;
+
+        return new Promise((resolve, reject) => {
+            assembler.getTranslations((error, translations) => {
+                if (error) {
+                    reject(error);
+                }
 
-Paper.prototype.loadTheme = function (paths, acceptLanguage, done) {
-    var self = this;
+                // Make translations available to the helpers
+                translator = Translator.create(data.acceptLanguage, translations);
+                let instance = new Paper(data.context.settings, data.context.themeSettings, assembler);
+                instance.renderingContext = data.context;
+                instance.translator = translator;
 
-    if (!_.isArray(paths)) {
-        paths = paths ? [paths] : [];
+                resolve(instance);
+            });
+        });
     }
 
-    Async.parallel([
-        function (next) {
-            self.loadTranslations(acceptLanguage, next);
-        },
-        function (next) {
-            Async.map(paths, self.loadTemplates.bind(self), next);
-        }
-    ], done);
-};
+    /**
+     * Renders a string with the given context
+     * @param  {String} string
+     * @param  {Object} context
+     */
+    renderString(string, context) {
+        return this.handlebars.compile(string)(context);
+    }
 
-/**
- * Load Partials/Templates
- * @param  {Object}   templates
- * @param  {Function} callback
- */
-Paper.prototype.loadTemplates = function (path, callback) {
-    var self = this;
-
-    var processor = self.getTemplateProcessor();
-
-    self.assembler.getTemplates(path, processor, function (error, templates) {
-        if (error) {
-            return callback(error);
+    loadTheme(paths, acceptLanguage, done) {
+        let self = this;
+
+        if (!_.isArray(paths)) {
+            paths = paths ? [paths] : [];
         }
 
-        _.each(templates, function (precompiled, path) {
-            var template;
-            if (!self.handlebars.templates[path]) {
-                eval('template = ' + precompiled);
-                self.handlebars.templates[path] = self.handlebars.template(template);
+        Async.parallel([
+            function (next) {
+                self.loadTranslations(acceptLanguage, next);
+            },
+            function (next) {
+                Async.map(paths, self.loadTemplates.bind(self), next);
             }
-        });
+        ], done);
+    }
 
-        self.handlebars.partials = self.handlebars.templates;
+    /**
+     * Load Partials/Templates
+     * @param  {Object}   templates
+     * @param  {Function} callback
+    */
+    loadTemplates(path, callback) {
+        let self = this;
```

Comment:
Same here. You don't need this anymore. Just make sure to convert all anonymous functions inside this block to use arrow functions